File: LeetCode/452.minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<pair<int, int>>& points) {
        int size = points.size();
        if (0 == size)  return 0;
        sort(points.begin(), points.end());
        int res = 1;
        int start = points[0].first, end = points[0].second;
        for (int i = 1; i < size; i++){
            if ((points[i].first >= start && points[i].second <= end)
                    || (points[i].first <= end && points[i].second >= end)){
                start = max(points[i].first, start);
                end = min(points[i].second, end);
            }else{
                res++;
                start = points[i].first;
                end = points[i].second;
            }
        }
        return res;
    }

};

File: LeetCode/452.minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        if (null == points || 0 == points.length)   return 0;
        int res = 1;
        Arrays.sort(points, new Comparator<int[]>() {
            @Override
            public int compare(int[] a, int[] b) {
                return a[1] - b[1];
            }
        });
        int end = points[0][1];
        for (int i = 1;i < points.length; i++){
            if (points[i][0] <= end)    continue;
            res++;
            end = points[i][1];
        }
        return res;
    }
}